src/Renci.SshNet/SftpClient.cs

```diff
@@ -2121,6 +2122,11 @@ public IAsyncResult BeginSynchronizeDirectories(string sourcePath, string destin
                 throw new ArgumentException("destDir");
```

Comment:
```suggestion
                throw new ArgumentException(nameof(destinationPath));
```

If it's not too much trouble while we are here :-)

Reply:
Strictly speaking this is (already) wrong since ArgumentException expects a message, not a parameter name. I changed it anyway.

ArgumentException.ThrowIfNullOrEmpty would be nice here, but it's only available in .NET 7+.